PWGHF/Utils/utilsMcMatching.h

```diff
@@ -0,0 +1,244 @@
+  {DecayChannelMain::LcToPPiPi, {+kProton, +kPiMinus, +kPiPlus}},
+  {DecayChannelMain::LcToPKK, {+kProton, +kKMinus, +kKPlus}}};
+
+// Xic → p K∓ π±
+static const std::unordered_map<DecayChannelResonant, const std::array<int, 2>> DaughtersXiCResonant{
+  {DecayChannelResonant::XicToPKstar0, {-o2::constants::physics::kK0Star892, +kProton}},
+  {DecayChannelResonant::XicToPPhi, {+kProton, +o2::constants::physics::kPhi}},
+};
+
+static const std::unordered_map<DecayChannelMain, const std::vector<int>> DaughtersXiCMain{
+  {DecayChannelMain::XicToPKPi, {+kProton, +kKMinus, +kPiPlus}},
+  {DecayChannelMain::XicToPKK, {+kProton, +kKMinus, +kKPlus}},
+  {DecayChannelMain::XicToSPiPi, {+kSigmaPlus, +kPiMinus, +kPiPlus}},
+};
+
+/// Returns a map of the possible final states for a specific 3-prong particle specie
+/// \param pdgMother PDG code of the mother particle
+/// \return a map of final states with their corresponding PDG codes
+inline std::unordered_map<DecayChannelMain, const std::vector<int>> getDecayChannel3Prong(int pdgMother)
+{
+  switch (pdgMother) {
+    case o2::constants::physics::Pdg::kDPlus:
+      return DaughtersDPlusMain;
+    case o2::constants::physics::Pdg::kDS:
+      return DaughtersDsMain;
+    case o2::constants::physics::Pdg::kDStar:
+      return DaughtersDstarMain;
+    case o2::constants::physics::Pdg::kLambdaCPlus:
+      return DaughtersLcMain;
+    case o2::constants::physics::Pdg::kXiCPlus:
+      return DaughtersXiCMain;
+    default:
+      LOG(error) << "Unknown PDG code for 3-prong final states: " << pdgMother;
+      return {};
+  }
+}
+
+/// Returns a map of the resonant decay channels for a specific 3-prong particle specie
+/// \param pdgMother PDG code of the mother particle
+/// \return a map of resonant decay channels with their corresponding PDG codes
+inline std::unordered_map<DecayChannelResonant, const std::array<int, 2>> getResoChannels3Prong(int pdgMother)
+{
+  switch (pdgMother) {
+    case o2::constants::physics::Pdg::kDPlus:
+      return DaughtersDPlusResonant;
+    case o2::constants::physics::Pdg::kDS:
+      return DaughtersDsResonant;
+    case o2::constants::physics::Pdg::kDStar:
+      return DaughtersDstarResonant;
+    case o2::constants::physics::Pdg::kLambdaCPlus:
+      return DaughtersLcResonant;
+    case o2::constants::physics::Pdg::kXiCPlus:
+      return DaughtersXiCResonant;
+    default:
+      LOG(error) << "Unknown PDG code for 3-prong final states: " << pdgMother;
+      return {};
+  }
+}
+
+} // namespace hf_cand_3prong
+
+/// Perform the matching for a single resonant channel
+/// \tparam N size of the array of daughter PDG codes
+/// \param arrPdgResoChn array of daughter indices
+/// \param arrPdgDaugs array of PDG codes for the resonant decay
+/// \return true if the resonant channel is matched, false otherwise
+template <std::size_t N>
+inline bool checkResonantDecay(std::array<int, N> const& arrPdgResoChn, std::array<int, N> arrPdgDaugs)
```

Comment:
There is nothing specific to resonant channels in this function which makes the name misleading.